packages/smart/contracts/rewards/RewardsManager.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.7.0;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "./MasterChef.sol";
+
+contract RewardsManager is Ownable {
+    using SafeMath for uint256;
+
+    modifier onlyApprovedMasterChefs() {
+        require(approvedMasterChefs[msg.sender], "Sender not approved for this action");
+        _;
+    }
+
+    ERC20 private rewardsToken;
+    mapping(address => bool) private approvedMasterChefs;
+
+    constructor(ERC20 _rewardsToken) Ownable() {
+        rewardsToken = _rewardsToken;
+    }
+
+    function trustMasterChef(address _masterChef) public onlyOwner {
+        approvedMasterChefs[_masterChef] = true;
+    }
+
+    function untrustMasterChef(address _masterChef) public onlyOwner {
+        approvedMasterChefs[_masterChef] = false;
```

Comment:
Side note: I believe this is identical to deletion. Do you happen to know for sure?

Reply:
https://ethereum.stackexchange.com/questions/58495/understanding-more-about-delete-keyword-in-solidity/58496

It appears `delete` has a slight gas advantage but very little practical difference.